tests/e2e/support/mock_client.rs

```diff
@@ -189,6 +198,20 @@ impl MockClient {
                     self.auto_respond_to_workspace_configuration_request(request);
                 }
             }
+            if let Message::Notification(Notification { method, params }) = &message {
+                match method.as_str() {
+                    "textDocument/publishDiagnostics" => {
+                        let params: PublishDiagnosticsParams =
+                            serde_json::from_value(params.clone())
+                                .expect("failed to parse `textDocument/publishDiagnostics` params");
+                        self.diagnostics.insert(params.uri, params.diagnostics);
+                    }
+                    "cairo/diagnosticsCalculationStart" => {
+                        self.diagnostics.clear();
+                    }
+                    _ => {}
```

Comment:
why are you doing this here? this is unlike the behaviour that editors seem to do, isn't it?

Reply:
I keep only last diagnostics generation to avoid maintaining complicated logic. We don't need that behavior in the tests anyway.